### PR TITLE
Gather etcd mutex profile and refactor profile.go

### PR DESCRIPTION
**Pull request by mm4tt**
Refactoring in profile.go allows adding new profile types much easier, it also removes duplicated and unnecessary code.

### CODE REVIEW

**Comment by mm4tt** [comment]
Ref. https://github.com/kubernetes/kubernetes/issues/80209

File: clusterloader2/pkg/measurement/common/profile.go [hunk 1]
```diff
@@ -75,13 +76,20 @@ type profileMeasurement struct {
...
+	return func() measurement.Measurement {
+		return &profileMeasurement{
+			name:   name,
+			config: &profileConfig{kind: kind},
+		}
+	}
+}
+
+func (p *profileMeasurement) start(config *measurement.MeasurementConfig) error {
+	err := p.populateProfileConfig(config)
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/profile.go]: if err := ...; err != nil {
  return err
}

**Comment by wojtek-t** [comment]
Also travis is failing - golint.

File: clusterloader2/pkg/measurement/common/profile.go (hunk 1 above)
Comment (mm4tt) [reply clusterloader2/pkg/measurement/common/profile.go]: Done.

**Pull request merged**